modules/post/windows/gather/credentials/vnc.rb

```diff
@@ -254,47 +242,50 @@ def run
 
     print_status("Enumerating VNC passwords on #{sysinfo['Computer']}")
 
-    locations.map do |e|
-      vprint_status("Checking #{e[:name]}...")
+    locations.map do |location|
+      vprint_status("Checking #{location[:name]}...")
+
       if e.key?(:check_reg)
-        e[:port] = reg_get(e[:check_reg], e[:port_variable])
-        e[:hash] = reg_get(e[:check_reg], e[:pass_variable])
-        e[:pass] = decrypt_hash(e[:hash])
-        if e.key?(:viewonly_variable)
-          e[:viewonly_hash] = reg_get(e[:check_reg], e[:viewonly_variable])
-          e[:viewonly_pass] = decrypt_hash(e[:viewonly_hash])
+        location[:port] = reg_get(location[:check_reg], location[:port_variable])
+        location[:hash] = reg_get(location[:check_reg], location[:pass_variable])
+        location[:pass] = decrypt_hash(location[:hash])
+        if location.key?(:viewonly_variable)
+          location[:viewonly_hash] = reg_get(location[:check_reg], location[:viewonly_variable])
+          location[:viewonly_pass] = decrypt_hash(location[:viewonly_hash])
         end
-      elsif e.key?(:check_file)
-        e[:port] = file_get(e[:check_file], e[:port_variable])
-        e[:hash] = file_get(e[:check_file], e[:pass_variable])
-        e[:pass] = decrypt_hash(e[:hash])
-        if e.key?(:viewonly_variable)
-          e[:viewonly_hash] = file_get(e[:check_file], e[:viewonly_variable])
-          e[:viewonly_pass] = decrypt_hash(e[:viewonly_hash])
+      elsif location.key?(:check_file)
+        location[:port] = file_get(location[:check_file], location[:port_variable])
+        location[:hash] = file_get(location[:check_file], location[:pass_variable])
+        location[:pass] = decrypt_hash(location[:hash])
+        if location.key?(:viewonly_variable)
+          location[:viewonly_hash] = file_get(location[:check_file], location[:viewonly_variable])
+          location[:viewonly_pass] = decrypt_hash(location[:viewonly_hash])
         end
       end
+
+      next if location[:pass].nil? && location[:viewonly_pass].nil?
+
+      location[:port] = 5900 if location[:port].nil?
+
       # reporting
+      service_data = {
+        address: ::Rex::Socket.getaddress(session.sock.peerhost, true),
+        port: location[:port].nil? ? 5900 : location[:port],
```

Comment:
Should this still be: `port: location[:port]` because above you've got ` location[:port] = 5900 if location[:port].nil?` now

Reply:
Correct. Fixed.